driver-core/src/main/com/mongodb/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -78,12 +86,23 @@ public void apply(final AsyncAggregateResponseBatchCursor<RawBsonDocument> curso
                 cursor.tryNext(callback);
                 cachePostBatchResumeToken(cursor);
             }
-        }, convertResultsCallback(callback));
+        }, convertResultsCallback(callback), true);
     }
 
     @Override
     public void close() {
-        if (!isClosed()) {
+        boolean closed;
```

Comment:
Can we simplify this in the same way that we did for AsyncQueryBatchCursor?

Reply:
Done